allure-framework-modules/mrchecker-database-module/src/main/java/com/capgemini/ntc/database/core/base/driver/DriverManager.java

```diff
@@ -0,0 +1,84 @@
+package com.capgemini.ntc.database.core.base.driver;
+
+import com.capgemini.ntc.database.core.base.properties.PropertiesFileSettings;
+import com.capgemini.ntc.database.core.base.runtime.RuntimeParameters;
+import com.capgemini.ntc.test.core.logger.BFLogger;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+
+public class DriverManager {
+
+	private static ThreadLocal<String> drivers = new ThreadLocal<String>();
+
+	private static PropertiesFileSettings propertiesFileSettings;
+
+	@Inject
+	public DriverManager(@Named("properties") PropertiesFileSettings propertiesFileSettings) {
+
+		if (null == DriverManager.propertiesFileSettings) {
+			DriverManager.propertiesFileSettings = propertiesFileSettings;
+		}
+
+		this.start();
+	}
+
+	public void start() {
+		DriverManager.getDriver();
+	}
+
+	public void stop() {
+		try {
+			closeDriver();
+			BFLogger.logDebug("Closing Driver in stop()");
+		} catch (Exception e) {
+		}
+	}
+
+	@Override
+	protected void finalize() throws Throwable {
+		super.finalize();
+		try {
+			closeDriver();
+			BFLogger.logDebug("Closed Driver in finalize()");
+		} catch (Exception e) {
+		}
+
+	}
+
+	public static String getDriver() {
+		String driver = drivers.get();
+		if (driver == null) {
+			driver = createDriver();
+			drivers.set(driver);
+			BFLogger.logDebug("driver:" + driver.toString());
+		}
+		return driver;
+	}
+
+	public static void closeDriver() {
+		String driver = drivers.get();
+		if (driver == null) {
+			BFLogger.logDebug("closeDriver() was called but there was no driver for this thread.");
+		} else {
+			try {
+				// driver.quit();
```

Comment:
you do not close db driver. Is it ok ?

Reply:
Just marked to do in the nearest future.